.github/workflows/go.yml

```diff
@@ -0,0 +1,22 @@
+name: Go
+
+on:
+  pull_request:
+    branches: [ main ]
```

Comment:
can we turn this on for all branches?

Reply:
@samira-barouti I don't think we need it for the moment, as we have no other branches we maintain and no new branches we're anticipating. In addition, In practice, `main` is the only protected branch. When we add branches, we'll have to determine how we handle those semantics (this, and protection - assuming we can protect multiples).

Reply:
Agreed. Also, it only applies to this particular repo. It would not, for instance, apply to a branch on your personal fork, if that's what you were thinking.

Reply:
Exactly! My intention was to turn this on for all branches to ensure these actions are running on the feature branches that the contributors push to the remote repo. 

Out of curiosity: is this doable at all with GH Actions, or we need a full-fledged CI like prow?

Reply:
I think as long as the PR is targeting main on this repo, it will run.